# Biomedical Databases Design, Implementation and Optimisation


## WS- 2017/2018


## Team 3 


# Starting Point

![question_mark](question_mark.jpg)

![disease_network](disease_network.jpg)

# Client Product

![Services_Dev](Services-Dev.png)

# Product Development Steps

* Choose the biomedical database and obtain raw data 

* Design and implement a data model in python

* Provide a python package that handles querying the database

* Query the database

* Document the project 

* Test the product

# Applications of Disease-Protein Database

## Find Associated Diseases with Given Disease  
Start with an OMIM ID for a particular disease in question and explore the diseases which seem to be related to the disease in question due to their overlap in associated proteins.  

![application_1](application_1.jpg)

In [1]:
#importing and populating the db
from biodb_team_3 import db_manager
m = db_manager.Manager()
m.populate_db()

C:\Users\admin\Anaconda3\lib\re.py:203: FutureWarning: split() requires a non-empty pattern match.
  return _compile(pattern, flags).split(string, maxsplit)


## Example
[Alzheimer Disease](https://www.omim.org/entry/104300) is one of the most prominent and debilitating forms of dementia, affecting millions of people world wide; its OMIM id is 104300.  
Query associated disease with this OMIM id:

In [11]:
associatedDisease = m.get_associated_disease_with_omim_id("104300")
print(associatedDisease)

{'605714'}


 The result, [Cerebral Amyloid Angiopathy](https://www.omim.org/entry/605714)(OMIM id: 605714), can be caused by the same amyloid protein that is associated with Alzheimer's dementia according to [Wikipedia](https://en.wikipedia.org/wiki/Cerebral_amyloid_angiopathy).

## Find Common Associated Proteins with Multiple Diseases
Another useful function is to find common associated proteins given a list of diseases. This might be helpful for finding the essential target for treating a patient with multiple diseases.  

![application_2](application_2.jpg)

### Application in Personalized Medicine

* Treatment based on all the concurrent diseases of a patient

* Find possible causes(proteins) for multiple concurrent diseases

* Verify the causes(check patient's protein expression profile, etc.)

* Reduce the amount of drug: Focus on the target protein and remove unnecessary non-targeting medicine

### Example

Here we use [Type 1 Diabetes](https://www.omim.org/entry/222100)(OMIM id: 222100) and several other diseases for query.  
  
  
First we query type 1 diabetes with [Celiac Disease](https://www.omim.org/entry/612011)(OMIM id: 612011)

In [3]:
commonProteins = m.get_shared_associated_proteins_by_omim_ids(["222100", "612011"])
print(commonProteins)

{'Q9UQQ2'}


The result, [SH2B Adapter Protein 3](http://www.uniprot.org/uniprot/Q9UQQ2)(UniProt id: Q9UQQ2) is involved in autoimmune disorders including type1 diabetes and celiac disease. 

Then we query with [Systemic Lupus Erythematosus](https://www.omim.org/entry/152700)(OMIM id: 152700), [Rheumatoid Arthritis](https://www.omim.org/entry/180300)(OMIM id: 180300) and [Vitiligo](https://www.omim.org/entry/193200)(OMIM id: 193200)

In [4]:
commonProteins = m.get_shared_associated_proteins_by_omim_ids(["222100", "152700"])
print(commonProteins)
commonProteins = m.get_shared_associated_proteins_by_omim_ids(["222100", "180300"])
print(commonProteins)
commonProteins = m.get_shared_associated_proteins_by_omim_ids(["222100", "193200"])
print(commonProteins)
commonProteins = m.get_shared_associated_proteins_by_omim_ids(["222100", "152700", "180300", "193200"])
print(commonProteins)

{'Q9Y2R2'}
{'Q9Y2R2'}
{'Q9Y2R2'}
{'Q9Y2R2'}


Any combination of the diseases above along with diabetes leads to the same result: [Tyrosine-protein Phosphatase Non-Receptor Type 22, or PTPN22](http://www.uniprot.org/uniprot/Q9Y2R2)(UniProt id: Q9Y2R2).   
  
  
According to [WikiPedia](https://en.wikipedia.org/wiki/PTPN22), the common 1858T [(rs2476601)](https://www.ncbi.nlm.nih.gov/projects/SNP/snp_ref.cgi?rs=2476601) Arg620Trp nonsynonymous single nucleotide polymorphism located in the PTPN22 gene has been associated with autoimmune disorders, including an increased risk of Type 1 Diabetes, rheumatoid arthritis, Systemic Lupus Erythematosus and Vitiligo.    
  
From these examples we can see that this query function might be of great help in diagnosing the target protein for multiple coexisting diseases.